cdap-ui/app/features/adapters/controllers/tabs/runs/tabs/status-ctrl.js

```diff
@@ -77,26 +81,22 @@ angular.module(PKG.name + '.feature.adapters')
       }
     };
 
-    datasrc.request(
+    datasrc.poll(
       {
         method: 'POST',
         _cdapPath: '/metrics/search?target=metric&' + tagQueryParams
-      })
-        .then(
-          function onMetricsDiscoverySuccess(res) {
-            widget.metric.names = res;
-            if (res.length > 0) {
-              pollForMetricsData(widget);
-            } else {
-              $scope.formattedData = [];
-            }
-          },
-          function onMetricsDiscoveryError() {
-            console.error('Error on Metrics fetch');
-          }
-        );
-    function pollForMetricsData(widget) {
-      DashboardHelper.pollData(widget)
+      },
+      function onMetricsDiscoverySuccess(res) {
+        widget.metric.names = res;
+        if (res.length > 0) {
+          FetchMetricsData(widget);
+        } else {
+          $scope.formattedData = [];
+        }
+      }
+    );
+    function FetchMetricsData(widget) {
```

Comment:
Any reason why this function starts with capital letter? We usually reserve capital letter for Objects/Factories right?